### PR TITLE
Use Leader ALOC if it is commissioner

### PR DESCRIPTION
**Pull request by lluwang**
This is to pass 9.2.4 (DUT as commissioner).

### CODE REVIEW

**Comment by chshu** [comment]
Thread specification doesn't distinguish on-mesh or external commissioner on this case, just ask to use Leader ALOC for the messages from commissioner to Leader. And I think it's ok to use Leader ALOC either Commissioner knows Leader RLOC or not, since all routers could help to map Leader ALOC to Leader RLOC.

**Comment by jwhui** [comment]
I think we should use Leader ALOC everywhere and not distinguish between Commissioner and non-Commissioner roles.

File: src/core/thread/meshcop_dataset_manager.cpp [hunk 1]
```diff
@@ -544,7 +557,20 @@ ThreadError DatasetManager::SendSetRequest(const otOperationalDataset &aDataset,
...
+    isCommissioner = mNetif.GetCommissioner().GetState() >= MeshCoP::Commissioner::kStateActive ? true : false;
+#endif
+
+    if (isCommissioner)
+    {
+        mMle.GetLeaderAloc(messageInfo.GetPeerAddr());
+    }
+    else
+    {
+        mMle.GetLeaderAddress(messageInfo.GetPeerAddr());
```
Comment (jwhui) [inline src/core/thread/meshcop_dataset_manager.cpp]: Let's just use the Leader ALOC all the time.  Reduces complexity.

File: src/core/thread/meshcop_dataset_manager.cpp [hunk 2]
```diff
@@ -595,7 +622,20 @@ ThreadError DatasetManager::SendGetRequest(const uint8_t *aTlvTypes, const uint8
...
```
Comment (jwhui) [inline src/core/thread/meshcop_dataset_manager.cpp]: Let's just use the Leader ALOC all the time.  Reduces complexity.

File: src/core/thread/network_data.cpp [hunk 3]
```diff
@@ -636,7 +638,20 @@ ThreadError NetworkData::SendServerDataNotification(uint16_t aRloc16)
...
+    isCommissioner = mNetif.GetCommissioner().GetState() >= MeshCoP::Commissioner::kStateActive ? true : false;
+#endif
+
+    if (isCommissioner)
+    {
+        mMle.GetLeaderAloc(messageInfo.GetPeerAddr());
+    }
+    else
+    {
+        mMle.GetLeaderAddress(messageInfo.GetPeerAddr());
```
Comment (jwhui) [inline src/core/thread/network_data.cpp]: Let's just use the Leader ALOC all the time.  Reduces complexity.

**Pull request merged**